### PR TITLE
Fix warnings and race condition in time keeping.

### PR DESCRIPTION
**Pull request by jaykrell**
```
/s/mono/mono/utils/mono-time.c:23:34: warning: unused variable 's_TimebaseInfo' [-Wunused-variable]
static mach_timebase_info_data_t s_TimebaseInfo;

/s/mono/mono/utils/mono-time.c:112:1: warning: unused function 'get_boot_time' [-Wunused-function]
```

### CODE REVIEW

File: mono/utils/mono-time.c [hunk 1]
```diff
@@ -109,6 +108,7 @@ mono_100ns_datetime (void)
 
 #include <time.h>
 
+#if 0
```
Comment (lambdageek) [inline mono/utils/mono-time.c]: Just delete it.   It should've been removed as part of 06a4273d73cfa732d911bdaf50f752c025a0665c

Comment (jaykrell) [reply mono/utils/mono-time.c]: Ok.

**Comment by jaykrell** [comment]
The struct is only two uint32. The  barrier could be eliminated, at least on 64bit platforms, by doing a 64bit write to it.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Pull request merged**